### PR TITLE
feat(cb2-7759): navigate to the summary page after creating a tech record

### PR DESCRIPTION
**Pull request by tahayazici**
## Navigate to the newly created Tech Record following creation

[CB2-7759](https://dvsa.atlassian.net/browse/CB2-7759)

### CODE REVIEW

File: src/app/models/vehicle-tech-record.model.ts [hunk 1]
```diff
@@ -8,18 +8,14 @@ export interface VehicleTechRecordModel {
...
 
-export interface postNewVehicleModel {
-  vin: string;
-  primaryVrm?: string;
-  trailerId?: string;
-  techRecord: TechRecordModel[];
-}
-
-export interface PutVehicleTechRecordModel extends Omit<VehicleTechRecordModel, 'vrms'> {
+export interface PostNewVehicleModel extends Omit<VehicleTechRecordModel, 'vrms' | 'trailerId'> {
```
Comment (gjulien-bjss) [inline src/app/models/vehicle-tech-record.model.ts:11]: We don't usually sent the partialVin? And we can sen trailerIds for trailers surely?

File: src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts [hunk 2]
```diff
@@ -1,38 +1,48 @@
...
   constructor(
+    private actions$: Actions,
     private globalErrorService: GlobalErrorService,
     private route: ActivatedRoute,
     private router: Router,
     private store: Store<TechnicalRecordServiceState>,
     private technicalRecordService: TechnicalRecordService
-  ) {}
+  ) {
+    this.actions$.pipe(ofType(createVehicleRecordSuccess), take(1), takeUntil(this.destroy$)).subscribe(({ vehicleTechRecords }) => {
```
Comment (gjulien-bjss) [inline src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts:33]: can't have both `take(1)` and `takeUntil()`

File: src/app/store/technical-records/effects/technical-record-service.effects.ts [hunk 3]
```diff
@@ -263,6 +268,21 @@ export class TechnicalRecordServiceEffects {
     )
   );
 
+  mapVehicleFromResponse(response: PostNewVehicleModel | PutVehicleTechRecordModel): VehicleTechRecordModel {
+    let trailerId: string | undefined;
+    let vrms: Vrm[] = [];
```
Comment (gjulien-bjss) [inline src/app/store/technical-records/effects/technical-record-service.effects.ts:273]:
```suggestion
    const vrms: Vrm[] = [];

```

File: src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts (hunk 2 above)
Comment (tahayazici) [reply src/app/features/tech-record/create/components/hydrate-new-vehicle-record/hydrate-new-vehicle-record.component.ts]: You're right, I've removed the `take(1)`.

File: src/app/models/vehicle-tech-record.model.ts (hunk 1 above)
Comment (tahayazici) [reply src/app/models/vehicle-tech-record.model.ts]: This is now corrected, thanks.

**Review by gjulien-bjss — APPROVED** [review]

**Pull request merged**